internal/provider/vaultsecrets/resource_vault_secrets_integration_confluent_test.go

```diff
@@ -0,0 +1,152 @@
+package vaultsecrets_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	secretmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/models"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/acctest"
+)
+
+func TestAccVaultSecretsResourceIntegrationConfluent(t *testing.T) {
+	cloudAPIKeyID := checkRequiredEnvVarOrFail(t, "CONFLUENT_API_KEY_ID")
+	cloudAPISecret := checkRequiredEnvVarOrFail(t, "CONFLUENT_API_SECRET")
+
+	integrationName1 := generateRandomSlug()
+	integrationName2 := generateRandomSlug()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create initial integration with access keys
+			{
+				Config: confluentConfig(integrationName1, cloudAPIKeyID, cloudAPISecret),
+				Check: resource.ComposeTestCheckFunc(
+					confluentCheckFuncs(integrationName1, cloudAPIKeyID, cloudAPISecret)...,
+				),
+			},
+			// Changing the name forces a recreation
+			{
+				Config: confluentConfig(integrationName2, cloudAPIKeyID, cloudAPISecret),
+				Check: resource.ComposeTestCheckFunc(
+					confluentCheckFuncs(integrationName2, cloudAPIKeyID, cloudAPISecret)...,
+				),
+			},
+			// Modifying mutable fields causes an update
+			{
+				Config: confluentConfig(integrationName2, cloudAPIKeyID, cloudAPISecret),
+				Check: resource.ComposeTestCheckFunc(
+					confluentCheckFuncs(integrationName2, cloudAPIKeyID, cloudAPISecret)...,
+				),
+			},
```

Comment:
The test does not pass a new value for a mutable field. It should either try to set a new API Key ID/Secret or be removed until we have a new mutable field to test against.